### PR TITLE
User can update a note by id.

### PR DESCRIPTION
**Pull request by danielduffield**
Created an event handler for PUT requests using the id parameter. Only allows PUT requests to alter the content or title of a note, sending a 404 if the request if neither key is contained with the request body.

### CODE REVIEW

**Comment by danielduffield** [comment]
screen shot 2017-07-06 at 10 54 25 am

**Comment by danielduffield** [comment]
screen shot 2017-07-06 at 10 57 14 am

File: index.js [hunk 1]
```diff
@@ -8,6 +8,17 @@ const notes = []
 
 app.use(jsonParser)
 
+app.put('/notes/:id', (req, res) => {
+  const noteId = parseInt(req.params.id, 10)
```
Comment (thebearingedge) [inline index.js:12]: What if you decide to define notes with more properties? Can you generalize this solution?

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: index.js [hunk 2]
```diff
@@ -10,11 +10,10 @@ app.use(jsonParser)
 
 app.put('/notes/:id', (req, res) => {
   const noteId = parseInt(req.params.id, 10)
-  if (req.body.content) {
-    notes[noteId - 1].content = req.body.content
-  }
-  if (req.body.title) {
-    notes[noteId - 1].title = req.body.title
+  for (const key in req.body) {
+    if (key === 'title' || key === 'content') {
```
Comment (thebearingedge) [inline index.js]: No, what if there's another key on notes tomorrow? Will you just keep adding `||`s?

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (danielduffield) [reply index.js]: Omitting the conditional would seemingly achieve that effect, but it seemed preferable to keep a conditional to regulate push requests

Comment (thebearingedge) [reply index.js]: What is a push request?

Comment (danielduffield) [reply index.js]: *put. I've made that mistake a couple times now, all these 'p' words

**Review by thebearingedge — CHANGES_REQUESTED** [review]
I tried to install your project, but there's no `package.json`.

File: index.js [hunk 3]
```diff
@@ -8,6 +9,16 @@ const notes = []
 
 app.use(jsonParser)
 
+app.put('/notes/:id', putValidator)
+
+app.put('/notes/:id', (req, res) => {
+  const noteId = parseInt(req.params.id, 10)
+  const note = notes.find(note => note.id === noteId)
+  if (!req.body) return res.sendStatus(404)
```
Comment (thebearingedge) [inline index.js]: This will never be `true`. Body parser ensures that there is at least an empty request body.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Pull request merged**